### PR TITLE
Progress Indicator: Adds `setup-assistant` variant

### LINKED ISSUE #1952: Vertical Progress Indicator: Support Setup Assistant
This needs some more clarity and more detail to meet definition of ready, but I'm thinking a `setup-assistant` variant be added to progress indicator that allows `actions` on right side and add the gray line divider, etc. This would then be used as a sub component of Setup Assistant. Since this would only be used in Setup Assistant at this point, I'd propose that it just be a simple `onRenderActions` render prop.

https://www.lightningdesignsystem.com/components/setup-assistant/#Hub-with-Expandable-Steps
and 
https://github.com/salesforce/design-system-react/issues/1884
for more information

### PR DESCRIPTION
**Pull request by pulkonet**
Fixes #1952

### Additional description

---

### CONTRIBUTOR checklist (do not remove)
Please complete for every pull request

* [ ] First-time contributors should sign the Contributor License Agreement. It's a fancy way of saying that you are giving away your contribution to this project. If you haven't before, wait a few minutes and a bot will comment on this pull request with instructions.
* [ ] `npm run lint:fix` has been run and linting passes.
* [ ] Mocha, Jest (Storyshots), and `components/component-docs.json` CI tests pass (`npm test`).
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).

### REVIEWER checklist (do not remove)

* [ ] TravisCI tests pass. This includes linting, Mocha, Jest, Storyshots, and `components/component-docs.json` tests.
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).
###### Required only if there are markup / UX changes
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [ ] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [ ] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

### CODE REVIEW

File: components/progress-indicator/private/step-vertical.jsx [hunk 1]
```diff
@@ -90,6 +90,32 @@ class StepVertical extends React.Component {
...
+			return (
+				<div
+					id={`progress-indicator-vertical-label-${this.props.step.id ||
+						this.props.index}`}
+					className="slds-progress__item_content slds-grid slds-grid_align-spread"
+				>
+					<div className="slds-size_3-of-4">{this.props.step.label}</div>
+					<div className="slds-grid slds-grid_align-end slds-size_1-of-4">
+						{this.props.step.onRenderSetupAssistantAction}
+					</div>
```
Comment (interactivellama) [inline components/progress-indicator/private/step-vertical.jsx]: Here <p class="slds-text-align_right slds-text-color_weak slds-p-top_medium">{prop}</p>

Comment (pulkonet) [reply components/progress-indicator/private/step-vertical.jsx]: Sure, I will put it here.

Comment (pulkonet) [reply components/progress-indicator/private/step-vertical.jsx]: Screenshot 2019-05-04 at 1 17 26 AM

@interactivellama Does this look fine?

**Comment by kevinparkerson** [comment]
thanks for this, it should be very helpful @pulkonet!

**Pull request merged**